src/main/java/net/wimpi/modbus/net/TCPConnectionHandler.java

```diff
@@ -80,8 +81,15 @@ public void run() {
                     }
                 }
 
-                logger.debug("Response (transaction id {}): {}", response.getTransactionID(), response.getHexMessage());
-                m_Transport.writeMessage(response);
+                try {
+                    m_Transport.writeMessage(response);
+                    logger.debug("Response (transaction id {}): {}", response.getTransactionID(), response.getHexMessage());
+                } catch (ModbusMessageSerializationException e) {
+                    logger.error("Error while serializing the response of request {}", request, e);
```

Comment:
Logging the wrapping `ModbusMessageSerializationException` does not add much value, as it is always used to wrap some other exception. I would recommend to unwrap the original exception and then log that.

Reply:
That will lose a bit of context, but I'm not against it. Done.